### PR TITLE
CVSL-2158 viewing and nav of eligibility checks

### PR DESCRIPTION
**Pull request by andrewrlee**

### CODE REVIEW

File: server/utils/nunjucksSetup.ts [hunk 1]
```diff
@@ -82,19 +89,9 @@ export function registerNunjucks(app?: express.Express): Environment {
     'dumpJson',
     (val: string) => new nunjucks.runtime.SafeString(`<pre>${JSON.stringify(val, null, 2)}</pre>`),
   )
-  njkEnv.addGlobal('paths', paths)
```
Comment (andrewrlee) [inline server/utils/nunjucksSetup.ts:85]: these had been added twice I think due to merge issues

File: server/services/eligiblityAndSuitabilityService.ts [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    token: string,
+    {
+      prisonNumber,
+      type,
+      check,
+      form,
+    }: { prisonNumber: string; type: string; check: Check; form: Record<string, unknown> },
+  ) {
+    const payload = this.sanitiseForm(check, form)
+    logger.info('save check called', prisonNumber, type, check, payload)
```
Comment (andrewrlee) [inline server/services/eligiblityAndSuitabilityService.ts:46]: not yet implemented writing back answers

File: server/services/eligiblityAndSuitabilityService.ts [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+  ) {
+    const initialChecks = await this.getInitialChecks(token, prisonNumber)
+    const checks = type === 'eligibility' ? initialChecks.eligibility : initialChecks.suitability
+    const checkIndex = checks.findIndex(q => q.code === checkCode)
+
+    assert(checkIndex >= 0, `Could not find check of type: '${type}', code: '${checkCode}'`)
+    return {
+      assessmentSummary: initialChecks.assessmentSummary,
+      check: checks[checkIndex],
+      nextCheck: checks[checkIndex + 1],
```
Comment (andrewrlee) [inline server/services/eligiblityAndSuitabilityService.ts:32]: nextCheck so we know where to nav to next

File: server/services/caseAdminCaseloadService.ts [hunk 4]
```diff
@@ -31,9 +31,4 @@ export default class CaseAdminCaseloadService {
     const assessForEarlyReleaseApiClient = this.assessForEarlyReleaseApiClientBuilder(token)
     return assessForEarlyReleaseApiClient.getAssessmentSummary(prisonNumber)
   }
-
```
Comment (andrewrlee) [inline server/services/caseAdminCaseloadService.ts:34]: extracted out separate service

File: server/middleware/setUpValidationMiddleware.ts [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+export const FLASH_KEY_VALIDATION_ERRORS = 'validationErrors'
+export const FLASH_KEY_SUBMITTED_FORM = 'submittedForm'
+
+export function validateRequest(request: Request, validator: Validator) {
+  const errors = validator(request.body)
+  if (errors.length) {
+    request.flash(FLASH_KEY_VALIDATION_ERRORS, errors)
+    request.flash(FLASH_KEY_SUBMITTED_FORM, request.body)
+    throw new ValidationError('A validation error occured')
+  }
```
Comment (andrewrlee) [inline server/middleware/setUpValidationMiddleware.ts:16]: Added a simple validation framework.

Validation works by:
* A custom validator function determines whether one or more errors needs reporting on.
* If so this function sets some flash state and throws a custom exception.
* The `errorHandler` detects this type of error then redirects back to the GET of the original request.
* Some middleware inspects flash state and adds any error message or form input to `res.locals`
* These can then be accessed in `nunjucks` templates

Calling this in a POST handler works nicely for the current use case - we don't have the required metadata to check whether the form data is valid until we're in the handler.


We could extend this to be called from a bit of middleware on a specific POST route for complex forms if we know the schema in advance and it makes sense 

I'd also like to add a custom macro wrapping the error summary bit in a follow up.

File: server/routes/__testutils/appSetup.ts [hunk 6]
```diff
@@ -59,7 +64,9 @@ function appSetup(services: Services, production: boolean, userSupplier: () => H
   })
   app.use(express.json())
   app.use(express.urlencoded({ extended: true }))
+  middlewareInjector(app)
```
Comment (andrewrlee) [inline server/routes/__testutils/appSetup.ts:67]: added this to allow testing a route throwing a validation error

File: server/routes/caseAdmin/initialChecks/check.ts [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+      checkCode,
+    )
+
+    validateRequest(req, body =>
+      check.questions.flatMap(question => {
+        const value = body?.[question.name]
+        const missingValue = value === undefined || (value !== 'true' && value !== 'false')
+        return missingValue ? [{ field: question.name, message: question.text }] : []
+      }),
+    )
```
Comment (andrewrlee) [inline server/routes/caseAdmin/initialChecks/check.ts:38]: performing validation.. for each question in this check make sure it's been answered

**Review by justin-gardiner — APPROVED** [review]
LGTM

**Pull request merged**